api/controller/assignments.js

```diff
@@ -0,0 +1,904 @@
+                                            $filter: {
+                                                input: "$content",
+                                                as: "item",
+                                                cond: { $eq: ["$$item.objectId", assignment.assignedWork] }
+                                            }
+                                        }
+                                    }
+                                },
+                            ]).toArray(function (err, entry) {
+                                if (err) {
+                                    return res.status(500).send({
+                                        'error': "An error has occurred",
+                                        'code': 10
+                                    });
+                                } else {
+                                    // adding assignment metadata to entry
+                                    if (entry.length > 0) {
+                                        entry[0].content[0].metadata.assignmentId = req.params.assignmentId;
+                                        entry[0].content[0].metadata.submissionDate = null;
+                                        entry[0].content[0].metadata.dueDate = assignment.dueDate;
+                                        entry[0].content[0].metadata.instructions = assignment.instructions;
+                                        entry[0].content[0].metadata.lateTurnIn = assignment.lateTurnIn;
+                                        entry[0].content[0].metadata.isSubmitted = false;
+                                        entry[0].content[0].metadata.status = null;
+                                        entry[0].content[0].metadata.comment = "";
+                                    } else {
+                                        return res.status(404).send({
+                                            'error': "Entry not found",
+                                            'code': 36
+                                        });
+                                    }
+                                    updateEntries(entry[0].content[0], privateJournalIds).then(function (result) {
+                                        updateStatus(req, res, "Assigned", entry[0].content[0].objectId, function (err, result) {
+                                            if (err) {
+                                                return res.status(500).send({
+                                                    'error': "An error has occurred",
+                                                    'code': 10
+                                                });
+                                            } else {
+                                                res.status(200).send(result);
+                                            }
+                                        });
+                                        res.status(200).send((result).toString());
```

Comment:
This looks like a mistake. This should not be there.

Reply:
I guess yes because there shouldn't be two res,send() with the same code